### PR TITLE
thread: Change the reallocation calculation and remove name from thread state

### PR DESCRIPTION
**Pull request by Aursen**

### CODE REVIEW

File: programs/thread/src/state/thread.rs [hunk 1]
```diff
@@ -73,30 +71,41 @@ pub trait ThreadAccount {
...
-            8,
-            size_of::<Thread>(),
-            self.id.len(),
-            self.instructions.try_to_vec()?.len(),
-            self.trigger.try_to_vec()?.len(),
-            NEXT_INSTRUCTION_SIZE,
-        ]
-        .iter()
-        .sum();
+        let data_len = 8 + size_of_val(self);
```
Comment (Adrena-Corto) [inline programs/thread/src/state/thread.rs:100]: What's the diff between size_of_val(self) and size_of(Self), first time I see it

Comment (Aursen) [reply programs/thread/src/state/thread.rs:100]: Returns the size of the pointed-to value in bytes

Comment (Aursen) [reply programs/thread/src/state/thread.rs:100]: https://doc.rust-lang.org/std/mem/fn.size_of_val.html

File: programs/thread/src/state/thread.rs [hunk 2]
```diff
@@ -73,30 +71,41 @@ pub trait ThreadAccount {
     fn realloc_account(&mut self) -> Result<()>;
 }
 
+impl Thread {
+    pub fn min_space(instructions_number: usize) -> usize {
```
Comment (Adrena-Corto) [inline programs/thread/src/state/thread.rs:75]: This is the bit I want to be using right?

Comment (Aursen) [reply programs/thread/src/state/thread.rs:75]: Yes exactly

File: programs/thread/src/state/thread.rs (hunk 1 above)
Comment (Aursen) [reply programs/thread/src/state/thread.rs:100]: I will try to vectorise and get the len we lost a bit of compute unit but it's more accurate

**Pull request merged**